assets/javascripts/vector-drag.js

```diff
@@ -0,0 +1,72 @@
+var ol = require('openlayers')
+
+function VectorDrag () {
+  ol.interaction.Pointer.call(this, {
+    handleDownEvent: VectorDrag.prototype.handleDownEvent,
+    handleDragEvent: VectorDrag.prototype.handleDragEvent,
+    handleMoveEvent: VectorDrag.prototype.handleMoveEvent,
+    handleUpEvent: VectorDrag.prototype.handleUpEvent
+  })
+
+  this.coordinate_ = null
+  this.cursor_ = 'pointer'
+  this.feature_ = null
+  this.previousCursor_ = undefined
+}
+
+ol.inherits(VectorDrag, ol.interaction.Pointer)
+
+VectorDrag.prototype.handleDownEvent = function (evt) {
+  var map = evt.map
+
+  var feature = map.forEachFeatureAtPixel(evt.pixel,
+     function (feature) {
+       return feature
+     })
+
+  if (feature) {
+    this.coordinate_ = evt.coordinate
+    this.feature_ = feature
+  }
+
+  return !!feature
+}
+
+VectorDrag.prototype.handleDragEvent = function (evt) {
+  var deltaX = evt.coordinate[0] - this.coordinate_[0]
+  var deltaY = evt.coordinate[1] - this.coordinate_[1]
+
+  var geometry = this.feature_.getGeometry()
+  geometry.translate(deltaX, deltaY)
+
+  this.coordinate_[0] = evt.coordinate[0]
+  this.coordinate_[1] = evt.coordinate[1]
+}
+
+VectorDrag.prototype.handleMoveEvent = function (evt) {
+  if (this.cursor_) {
+    var map = evt.map
+    var feature = map.forEachFeatureAtPixel(evt.pixel,
+     function (feature) {
+       return feature
+     })
+    var element = evt.map.getTargetElement()
+    if (feature) {
+      if (element.style.cursor !== this.cursor_) {
+        this.previousCursor_ = element.style.cursor
+        element.style.cursor = this.cursor_
+      }
+    } else if (this.previousCursor_ !== undefined) {
```

Comment:
Don't use undefined. Change to either

else if (typeof this._previousCursor !== 'undefined')

or simply

else if (this._previousCursor)